pkg/controller/toolchainstatus/creation_test.go

```diff
@@ -0,0 +1,54 @@
+package toolchainstatus
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/codeready-toolchain/host-operator/pkg/apis"
+	"github.com/codeready-toolchain/host-operator/pkg/configuration"
+	. "github.com/codeready-toolchain/host-operator/test"
+	. "github.com/codeready-toolchain/toolchain-common/pkg/test"
+	"github.com/stretchr/testify/require"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestCreateOrUpdateResources(t *testing.T) {
+	// given
+	s := scheme.Scheme
+	err := apis.AddToScheme(s)
+	require.NoError(t, err)
+	name := configuration.DefaultToolchainStatusName
+
+	t.Run("creation", func(t *testing.T) {
+		// given
+		cl := NewFakeClient(t)
+
+		// when
+		err = CreateOrUpdateResources(cl, s, HostOperatorNs, name)
+
+		// then
+		require.NoError(t, err)
+
+		// check that the MemberStatus exists now
+		AssertThatToolchainStatus(t, HostOperatorNs, name, cl).Exists()
+	})
+
+	t.Run("should return an error if creation fails ", func(t *testing.T) {
+		// given
+		cl := NewFakeClient(t)
+		cl.MockCreate = func(ctx context.Context, obj runtime.Object, opts ...client.CreateOption) error {
+			return fmt.Errorf("creation failed")
+		}
+
+		// when
+		err = CreateOrUpdateResources(cl, s, HostOperatorNs, name)
+
+		// then
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "unable to create resource")
```

Comment:
In the unit tests the message is `unable to create resource of kind: , version: : creation failed` because we don't set the `TypeMeta` here: https://github.com/rajivnathan/host-operator/blob/f5bfffbea9adeeb7c5c6956b674f2be5b70314dc/pkg/controller/toolchainstatus/creation.go#L14

But we don't set the TypeMeta for other similar places. eg: https://github.com/rajivnathan/host-operator/blob/f5bfffbea9adeeb7c5c6956b674f2be5b70314dc/pkg/controller/registrationservice/creation.go#L24

If I add the following then it becomes `unable to create resource of kind: ToolchainStatus, version: v1alpha1: creation failed` but then we need to remember to explicitly update the APIVersion string when we update the API version in the future.
```
TypeMeta: v1.TypeMeta{
	Kind:             "ToolchainStatus",
	APIVersion: "v1alpha1",
},
```

So maybe better to leave it as is?

Reply:
IMO that's OK if we check that the message is `unable to create resource of kind: , version: : creation failed`. The main point here is to make sure we do include the original error `creation failed`.

Reply:
Sounds good, I made the change here and for the other status-related unit tests as well